skimage/util/_backends.py

```diff
@@ -0,0 +1,149 @@
+import functools
+from importlib.metadata import entry_points
+from functools import lru_cache
+import os
+import warnings
+
+
+def _entry_points(group):
+    # Support Python versions before 3.10, which do not let you
+    # filter groups directly.
+    all_entry_points = entry_points()
+    if hasattr(all_entry_points, "select"):
+        selected_entry_points = all_entry_points.select(group=group)
+    else:
+        selected_entry_points = all_entry_points.get(group, ())
+    return selected_entry_points
+
+
+def disable_dispatching():
+    """Determine if dispatching has been disabled by the user"""
+    no_dispatching = os.environ.get("SKIMAGE_NO_DISPATCHING", False)
+    if no_dispatching == "1":
+        return True
+    else:
+        return False
+
+
+def get_module_name(func):
```

Comment:
@lagru this is my current best solution for the "which name should we use and how to automatically determine it".

It only works because scikit-image doesn't have sub-submodules in its public API. I think using something that only works in scikit-image is fine, especially because it makes this super simple compared to other possible solutions.

A thing that is missing IMHO is a way to test that the statement "scikit-image does not use sub-submodules in its public API" remains true/test breaks when it stops being true. Is there a way to iterate over all the public functions that scikit-image has (in scikit-learn we have `all_estimators()` that gives you all estimators)? If yes then we could extend the test I added.

Another approach is based on a suggestion from @seberg: make the `_dispatchable` decorator keep track of all the things it has decorated, then iterate over those in a test to make sure that they can be found at the name we generated.

Reply:
> It only works because scikit-image doesn't have sub-submodules in its public API. 

Unfortunately, we do have one sub-submodules: [`skimage.filters.rank`](https://scikit-image.org/docs/dev/api/skimage.filters.rank.html).

Reply:
> Is there a way to iterate over all the public functions that scikit-image has

Not that I'm aware of. Sometime ago I did some work to crawl our public API at runtime in https://github.com/scikit-image/scikit-image/pull/6905. But that's a bit too clunky to include in a test.

Reply:
> Unfortunately, we do have one sub-submodule: [`skimage.filters.rank`](https://scikit-image.org/docs/dev/api/skimage.filters.rank.html).

I'll add some code to handle this special case. Still think having this is better (because simpler) than trying to be smarter by inspecting modules. In particular because the code runs while the module is being imported

Reply:
I'll mark this as resolved for now, we can reopen it or move the discussion to #7550